File: backend/models/tab.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Tab struct {
	ID     primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	Groups []TabGroup         `json:"groups" bson:"groups"`
	Tempo  int                `json:"tempo" bson:"tempo"`
	Capo   int                `json:"capo" bson:"capo"`
}

type TabGroup struct {
	ID         primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	TabID      string             `json:"tabId" bson:"tabId"`
	GroupIndex int                `json:"groupIndex" bson:"groupIndex"`
	Notes      [][]Note           `json:"notes" bson:"notes"`
}

type Note struct {
	ID               primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	TabGroupID       string             `json:"tabGroupId,omitempty" bson:"tabGroupId,omitempty"`
	StringIndex      int                `json:"stringIndex" bson:"stringIndex"`
	Fret             string             `json:"fret" bson:"fret"`
	Position         int                `json:"position" bson:"position"`
	AbsolutePosition int                `json:"absolutePosition" bson:"absolutePosition"`
	Type             string             `json:"type,omitempty" bson:"type,omitempty"`
}

File: frontend/app/models/tab.ts
export interface Tab {
    _id: string;
    title?: string;
    groups: TabGroup[];
    tempo: number;
    capo: number;
    createdBy?: string;
    createdAt?: string;
    updatedAt?: string;
}

export interface TabGroup {
    _id: string;
    tabId: string;
    groupIndex: number;
    notes: Note[][];
}

export interface Note {
    _id: string;
    tabGroupId?: string;
    stringIndex: number;
    fret: string;
    position: number;
    absolutePosition: number;
    type?: 'h' | 'p' | 'space';
}

